#ifndef CAN_IDS_H
#define CAN_IDS_H

#define HYDROGREEN_CAN_BAUD_RATE    500000

typedef enum can_id_t {
    CAN_ID_TIME = 0x100,
    CAN_ID_TIME_BEFORE_TRANSMIT = 0x101,

    CAN_ID_ACC_BATTERY_VOLTAGE = 0x102,
    CAN_ID_ACC_BATTERY_CURRENT = 0x103,

    CAN_ID_FC_OUTPUT_VOLTAGE = 0x104,
    CAN_ID_FC_OUTPUT_CURRENT = 0x105,

    CAN_ID_SC_VOLTAGE = 0x106,
    CAN_ID_SC_CURRENT = 0x107,

    CAN_ID_MC_SUPPLY_VOLTAGE = 0x108,
    CAN_ID_MC_SUPPLY_CURRENT = 0x109,

    CAN_ID_FC_ENERGY_ACCUMULATED = 0x10A,

    CAN_ID_H2_PRESSURE_LOW = 0x10B,
    CAN_ID_H2_PRESSURE_FUEL_CELL = 0x10C,
    CAN_ID_H2_PRESSURE_HIGH = 0x10D,
    CAN_ID_H2_LEAKAGE_SENSOR_VOLTAGE = 0x10E,

    CAN_ID_FAN_DUTY_CYCLE = 0x10F,
    CAN_ID_BLOWER_DUTY_CYCLE = 0x110,

    CAN_ID_TEMP_FC_LOC1 = 0x111,
    CAN_ID_TEMP_FC_LOC2 = 0x112,

    CAN_ID_ACCEL_PEDAL_VOLTAGE = 0x113,
    CAN_ID_BRAKE_PEDAL_VOLTAGE = 0x114,
    CAN_ID_ACCEL_OUTPUT_VOLTAGE = 0x115,
    CAN_ID_BRAKE_OUTPUT_VOLTAGE = 0x116,

    CAN_ID_BUTTONS_MASTER_MASK = 0x117,
    CAN_ID_BUTTONS_STEERING_MASK = 0x118,

    CAN_ID_SENSOR_RPM = 0x119,
    CAN_ID_SENSOR_SPEED = 0x11A,

    CAN_ID_LAP_NUMBER = 0x11B,
    CAN_ID_LAP_TIME = 0x11C,

    CAN_ID_GPS_ALTITUDE = 0x11D,
    CAN_ID_GPS_LATITUDE = 0x11E,
    CAN_ID_GPS_LONGITUDE = 0x11F,
    CAN_ID_GPS_SPEED = 0x120,

    CAN_ID_MASTER_STATE = 0x121,
    CAN_ID_PROTIUM_STATE = 0x122,

    CAN_ID_MAIN_VALVE_ENABLE_OUTPUT = 0x123,
    CAN_ID_MC_ENABLE_OUTPUT = 0x124,

    CAN_ID_BUTTONS_LIGHTS_MASK = 0x125,
} can_id_t;

#endif
